src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -27,7 +27,7 @@ public class Settings : ISettings
         /// NuGet config names with casing ordered by precedence.
         /// </summary>
         public static readonly string[] OrderedSettingsFileNames =
-            (RuntimeEnvironmentHelper.IsWindows || RuntimeEnvironmentHelper.IsMacOSX) ?
+            RuntimeEnvironmentHelper.IsWindows ?
```

Comment:
This change would work, but it would be more efficient to use `PathUtility.IsFileSystemCaseInsensitive` to determine if the extra checks are actually needed. Could you use that instead? That utility didn't exist back when this settings code was originally created.